search/search_test.go

```diff
@@ -370,7 +370,9 @@ func TestTarjan(t *testing.T) {
 	for i, test := range tarjanTests {
 		g := concrete.NewDirectedGraph()
 		for u, e := range test.g {
-			g.AddNode(concrete.Node(u))
```

Comment:
No, it may be the case that an edge has been created to a node that has not yet been seen due to the test.g range loop. For example in the first Tarjan test case:

```
g: []set{
	0: linksTo(1),
	1: linksTo(2, 7),
	2: linksTo(3, 6),
	3: linksTo(4),
	4: linksTo(2, 5),
	6: linksTo(3, 5),
	7: linksTo(0, 6),
},
```

The first edge {0,1} creates the node 1, so by the time we get to adding node 1 explicitly, it is already there.

The API allows creation of both nodes implicitly during an `AddDirectedEdge` call and I will probably change these and others to that later, but here I want to be explicit until the API redesign has been more fully fleshed out (see #45 in particular).